src/eth_provider/provider.rs

```diff
@@ -552,27 +552,11 @@ where
         let filter = into_filter("tx.hash", &transaction_signed.hash, HASH_HEX_STRING_LEN);
         let pending_transaction = self.database.get_one::<StoredPendingTransaction>(filter.clone(), None).await?;
 
+        // Number of retries for the transaction (0 if it's a new transaction)
+        let retries = pending_transaction.as_ref().map(|tx| tx.retries).unwrap_or_default();
+
         // Determine the maximum fee
-        let max_fee = if cfg!(feature = "hive") {
-            u64::MAX
-        } else {
-            // TODO(Kakarot Fee Mechanism): When we no longer need to use the Starknet fees, remove this line.
-            // We need to get the balance (in Kakarot/Starknet native Token) of the signer to compute the Starknet maximum `max_fee`.
-            // We used to set max_fee = u64::MAX, but it'll fail if the signer doesn't have enough balance to pay the fees.
-            let eth_fees_per_gas =
-                transaction_signed.effective_gas_price(Some(transaction_signed.max_fee_per_gas() as u64)) as u64;
-            let eth_fees = eth_fees_per_gas.saturating_mul(transaction_signed.gas_limit());
-            let balance = self.balance(signer, None).await?;
-            let max_fee: u64 = balance.try_into().unwrap_or(u64::MAX);
-            let max_fee = (u128::from(max_fee) * 80 / 100) as u64;
-
-            // We add the retry count to the max fee in order to bypass the
-            // DuplicateTx error in Starknet, which rejects incoming transactions
-            // with the same hash. Incrementing the max fee causes the Starknet
-            // hash to change, allowing the transaction to pass.
-            let retries = pending_transaction.as_ref().map(|tx| tx.retries + 1).unwrap_or_default();
-            max_fee.saturating_sub(eth_fees).saturating_add(retries)
-        };
+        let max_fee = if cfg!(feature = "hive") { u64::MAX } else { u64::MIN };
```

Comment:
why u64::MIN? never seen this
just zero

Reply:
I have seen this on other repos, it's like using u64::MAX, u64::MIN will always be 0.

Reply:
Why wouldn't `0` be always 0?

Reply:
Also, this means that this PR is a double PR:
- enshrines 0 fee strategy in RPC
- retry service

-> need to document this strategy of 0 fee, maybe as a comment above the let max_fee

Reply:
> Why wouldn't `0` be always 0?

I mean it's just a constant, like you can use u64::MAX for 18446744073709551615, you can use u64::MIN for 0

Reply:
but writing `0` will infer type and be more intuitive no? just a nit